executor/restore_test.go

```diff
@@ -2,306 +2,270 @@ package executor
+		GivenJobConfig      *job.Config
+		GivenResource       *k8upv1a1.Restore
+		ExpectedPVC         PVCExpectation
+		ExpectedVolumeMount VolumeMountExpectation
+	}{
+		"givenS3RestoreResource_whenVolumeConfig_expectNoPVCAndNoMount": {
+			GivenJobConfig: newConfig(),
+			GivenResource:  newS3RestoreResource(),
+		},
+		"givenFolderRestoreResource_whenVolumeConfig_expectPVCAndMount": {
+			GivenJobConfig: newConfig(),
+			GivenResource:  newFolderRestoreResource(),
+			ExpectedPVC: PVCExpectation{
+				Expected:  true,
+				ClaimName: "test",
+				ReadOnly:  false,
+			},
+			ExpectedVolumeMount: VolumeMountExpectation{
+				Expected: true,
+				Name:     "test",
+				Path:     "/restore",
+			},
+		},
+	}
 
-	env["HOSTNAME"] = MatchFields(IgnoreExtras, Fields{
-		"Name": Equal("HOSTNAME"),
-	})
-	env["STATS_URL"] = MatchFields(IgnoreExtras, Fields{
-		"Name": Equal("STATS_URL"),
-	})
-	env["RESTIC_PASSWORD"] = MatchFields(IgnoreExtras, Fields{
-		"Name": Equal("RESTIC_PASSWORD"),
-	})
-	if _, ok := env["RESTIC_REPOSITORY"]; !ok {
-		env["RESTIC_REPOSITORY"] = MatchFields(IgnoreExtras, Fields{
-			"Name":  Equal("RESTIC_REPOSITORY"),
-			"Value": Equal("s3:/"),
+	for name, tt := range tests {
+		t.Run(name, func(t *testing.T) {
+			e := NewRestoreExecutor(*tt.GivenJobConfig)
+			volumes, mounts := e.volumeConfig(tt.GivenResource)
+
+			assertVolumes(t, tt.ExpectedPVC, volumes)
+			assertVolumeMonts(t, tt.ExpectedVolumeMount, mounts)
 		})
 	}
+}
+
+func assertVolumeMonts(t *testing.T, ex VolumeMountExpectation, mounts []corev1.VolumeMount) {
+	if !ex.Expected {
+		assert.Len(t, mounts, 0)
+		return
+	}
+
+	assert.Len(t, mounts, 1)
+	volumeMount := mounts[0]
 
-	return MatchFields(IgnoreExtras, Fields{
-		"ObjectMeta": MatchFields(IgnoreExtras, Fields{
-			"Labels": MatchAllKeys(Keys{
-				"k8upjob":           Equal("true"),
-				"k8upjob/exclusive": Equal("false"),
-				"k8up.io/type":      Equal("restore"),
-			}),
-		}),
-		"Spec": MatchFields(IgnoreExtras, Fields{
-			"Template": MatchFields(IgnoreExtras, Fields{
-				"Spec": MatchFields(IgnoreExtras, Fields{
-					"Volumes": MatchAllElements(volumeID, volumes),
-					"Containers": MatchAllElements(containerID, Elements{
-						"quay.io/vshn/wrestic:latest" + strings.Join(additionalArgs, ","): MatchFields(IgnoreExtras, Fields{
-							"Image":        Equal("quay.io/vshn/wrestic:latest"),
-							"Args":         ContainElements(additionalArgs),
-							"Env":          MatchAllElements(envID, env),
-							"VolumeMounts": MatchAllElements(volumeMountID, volumeMounts),
-						}),
-					}),
-				}),
-			}),
-		}),
-	})
+	assert.Equal(t, ex.Name, volumeMount.Name)
+	assert.Equal(t, ex.Path, volumeMount.MountPath)
 }
 
-func newConfig() *job.Config {
-	cfg := job.NewConfig(context.TODO(), nil, nil, &k8upv1alpha1.Restore{}, scheme, "")
-	return &cfg
+func assertVolumes(t *testing.T, ex PVCExpectation, volumes []corev1.Volume) {
+	if !ex.Expected {
+		assert.Len(t, volumes, 0)
+		return
+	}
```

Comment:
I don't see any tests now that use this flag. Maybe we can remove it?

Reply:
`givenS3RestoreResource_whenVolumeConfig_expectNoPVCAndNoMount` implicitly sets them `false`. I've now made this explicit.